File: 0001-0500/0260.Single Number III.py
class Solution(object):
    def singleNumber(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        seen = set()
        for num in nums:
            if num in seen:
                seen.discard(num)
            else:
                seen.add(num)
        return list(seen)

File: 0001-0500/0260.Single Number III.ts
function singleNumber(nums: number[]): number[] {
    const xor = nums.reduce((xor, num) => xor ^ num);
    const bit = xor & -xor;
    return nums.reduce((xors, num) => [xors[0] ^ (num & bit ? num : 0), xors[1] ^ (num & bit ? 0 : num)], [0, 0]);
};

